src/daft-ir/src/proto/rex.rs

```diff
@@ -288,6 +290,10 @@ impl ToFromProto for ir::Expr {
                 // todo(conner)
                 not_implemented_err!("exists")
             }
+            Self::ScalarFunc(ScalarFunc::Python(_)) => {
+                // todo
+                not_implemented_err!("python udf")
+            }
```

Comment:
The mapping between expressions and protos is kind of weird, maybe @rchowell you could give some guidance here.

There's currently a `PyFunction` and a `RsFunction` that both are variants of `Function`. Would it make sense to create a new `PyScalarFunction` variant here to hold the info of a PythonScalarFunc?

Reply:
Interestingly the proto type system is quite similar to the rust type system, albeit the tagged unions are more explicit. I did a best attempt at simplifying the function modeling in the protos, which is very similar to how I'm advising to simplify the type modeling in daft. The juxtaposition of our current DSL types and mapping those to the protos is indeed weird and shows some of the challenges of our current modeling.

It would make sense to (1) update Function -> ScalarFunction for consistency, and (2) make PyFunction a union with legacy/row/batch variants.